File: articles/institutional-norms-social-expectations/go/institutional_norms_score.go
package main

import "fmt"

type NormCase struct {
	NormRepetition          float64
	ExpectationConvergence  float64
	Internalization         float64
	SocialEnforcement       float64
	LegitimacyAlignment     float64
	TrustReinforcement      float64
	RoleClarity             float64
	LearningCapacity        float64
	FragmentationPressure   float64
	UnequalNormativeBurden  float64
	SuppressivePressure     float64
}

func NormativeStabilityRaw(x NormCase) float64 {
	return 0.13*x.NormRepetition +
		0.14*x.ExpectationConvergence +
		0.13*x.Internalization +
		0.11*x.SocialEnforcement +
		0.13*x.LegitimacyAlignment +
		0.11*x.TrustReinforcement +
		0.09*x.RoleClarity +
		0.08*x.LearningCapacity -
		0.13*x.FragmentationPressure -
		0.10*x.UnequalNormativeBurden -
		0.08*x.SuppressivePressure
}

func main() {
	demo := NormCase{82, 84, 78, 76, 80, 79, 77, 74, 22, 20, 18}
	fmt.Printf("Normative stability raw score: %.2f\n", NormativeStabilityRaw(demo))
}

File: articles/institutional-norms-social-expectations/rust/institutional_norms_score.rs
struct NormCase {
    norm_repetition: f64,
    expectation_convergence: f64,
    internalization: f64,
    social_enforcement: f64,
    legitimacy_alignment: f64,
    trust_reinforcement: f64,
    role_clarity: f64,
    learning_capacity: f64,
    fragmentation_pressure: f64,
    unequal_normative_burden: f64,
    suppressive_pressure: f64,
}

fn normative_stability_raw(x: &NormCase) -> f64 {
    0.13 * x.norm_repetition
        + 0.14 * x.expectation_convergence
        + 0.13 * x.internalization
        + 0.11 * x.social_enforcement
        + 0.13 * x.legitimacy_alignment
        + 0.11 * x.trust_reinforcement
        + 0.09 * x.role_clarity
        + 0.08 * x.learning_capacity
        - 0.13 * x.fragmentation_pressure
        - 0.10 * x.unequal_normative_burden
        - 0.08 * x.suppressive_pressure
}

fn main() {
    let demo = NormCase {
        norm_repetition: 82.0,
        expectation_convergence: 84.0,
        internalization: 78.0,
        social_enforcement: 76.0,
        legitimacy_alignment: 80.0,
        trust_reinforcement: 79.0,
        role_clarity: 77.0,
        learning_capacity: 74.0,
        fragmentation_pressure: 22.0,
        unequal_normative_burden: 20.0,
        suppressive_pressure: 18.0,
    };

    println!(
        "Normative stability raw score: {:.2}",
        normative_stability_raw(&demo)
    );
}
